### PR TITLE
build and push container

### PR DESCRIPTION
**Pull request by keiranjprice101**
## Description
Build the container on push to master

### CODE REVIEW

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.WRITE_PACKAGE_TOKEN }}
```
Comment (keiranjprice101) [inline .github/workflows/build-and-push.yml:29]: @Pasarus It is this token I am unsure of

File: .github/workflows/build-and-push.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+name: Build and Push Docker Images
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository_owner }}
```
Comment (Pasarus) [inline .github/workflows/build-and-push.yml:11]: Presently the images are named "rundetection": https://github.com/orgs/interactivereduction/packages/container/package/rundetection

Comment (keiranjprice101) [reply .github/workflows/build-and-push.yml:11]: The Image name var just sets the `interactivereduction` part of the name (I basically stole this from jobcontroller). I have removed the hyphen in the below part though.

Comment (Pasarus) [reply .github/workflows/build-and-push.yml:11]: Ah ok both of these work then

File: .github/workflows/dependency-review.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@ jobs:
       - name: Harden Runner
         uses: step-security/harden-runner@8ca2b8b2ece13480cda6dacd3511b49857a23c09 # v2.5.1
         with:
-          egress-policy: block
+          egress-policy: audit
```
Comment (keiranjprice101) [inline .github/workflows/dependency-review.yml:20]: Not sure why, but the allow list was reset, so had to move this back to audit. (It was blocking connections to github)

**Review by Pasarus — APPROVED** [review]

**Pull request merged**